### PR TITLE
Use correct origin point for calculating the absolute position on Android

### PR DESCRIPTION
**Pull request by j-piasecki**
## Description

In https://github.com/software-mansion/react-native-gesture-handler/pull/1812 I changed how absolute position is calculated to be relative to the window instead of the screen, but there were two things wrong with it:
- It used [getWindowVisibleDisplayFrame](https://developer.android.com/reference/android/view/View#getWindowVisibleDisplayFrame(android.graphics.Rect)) which `tells you the available area where content can be placed and remain visible to users`. This was always taking status bar height into account, even if the app could be drawn behind it.
- It was calling this method on the `decorView`, which is drawn under system bars anyway

This PR changes it to use `contentView` of the current activity and [getLocationOnScreen](https://developer.android.com/reference/android/view/View#getLocationOnScreen(int[])) which `gets the coordinates of this view in the coordinate space of the device screen, irrespective of system decorations and whether the system is in multi-window mode`.

In a nutshell, after this PR the absolute position will take into account whether the app is displayed behind the status bar.

## Test plan

Tested on the example app:

https://github.com/software-mansion/react-native-gesture-handler/assets/21055725/c3230a2d-7bba-483b-b160-1c9fb32b0da1

https://github.com/software-mansion/react-native-gesture-handler/assets/21055725/701d9fa0-0c9f-4126-af58-f80c6a0a6526

### CODE REVIEW

File: android/src/main/java/com/swmansion/gesturehandler/core/GestureHandler.kt [hunk 1]
```diff
@@ -192,10 +188,11 @@ open class GestureHandler<ConcreteGestureHandlerT : GestureHandler<ConcreteGestu
...
   protected open fun onPrepare() {}
 
-  private fun getWindow(context: Context?): Window? {
+  private fun getActivity(context: Context?): Activity? {
     if (context == null) return null
-    if (context is Activity) return context.window
-    if (context is ContextWrapper) return getWindow(context.baseContext)
+    if (context is ReactContext) return context.currentActivity
+    if (context is Activity) return context
+    if (context is ContextWrapper) return getActivity(context.baseContext)
```
Comment (m-bert) [inline android/src/main/java/com/swmansion/gesturehandler/core/GestureHandler.kt:195]:
```suggestion
    return when(context){
      is ReactContext -> context.currentActivity
      is Activity -> context
      is ContextWrapper -> getActivity(context.baseContext)
      else -> null
    }
```

Wouldn't this work? I think it looks cleaner.

**Review by m-bert — APPROVED** [review]
LGTM

**Comment by m-bert** [comment]
Also you can add that it fixes #2816

File: android/src/main/java/com/swmansion/gesturehandler/core/GestureHandler.kt (hunk 1 above)
Comment (j-piasecki) [reply android/src/main/java/com/swmansion/gesturehandler/core/GestureHandler.kt]: Done in https://github.com/software-mansion/react-native-gesture-handler/pull/2826/commits/422aa713724fc9dfad11757af2259c5cad7b2989
